builtin/packer/config_sourcer.go

```diff
@@ -0,0 +1,235 @@
+package packer
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-hclog"
+	packer "github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/stable/2021-04-30/client/packer_service"
+	hcpconfig "github.com/hashicorp/hcp-sdk-go/config"
+	"github.com/hashicorp/hcp-sdk-go/httpclient"
+	"github.com/mitchellh/mapstructure"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/component"
+	"github.com/hashicorp/waypoint-plugin-sdk/docs"
+	pb "github.com/hashicorp/waypoint-plugin-sdk/proto/gen"
+)
+
+type ConfigSourcer struct {
+	config sourceConfig
+	client packer.Client
+}
+
+type sourceConfig struct {
+	ClientId       string `hcl:"client_id,optional"`
+	ClientSecret   string `hcl:"client_secret,optional"`
+	OrganizationId string `hcl:"organization_id,attr"`
+	ProjectId      string `hcl:"project_id,attr"`
+}
+
+type reqConfig struct {
+	Bucket  string `hcl:"bucket,attr"`
+	Channel string `hcl:"channel,attr"`
+	Region  string `hcl:"region,attr"`
+	Cloud   string `hcl:"cloud,attr"`
+}
```

Comment:
Both this and the values in `sourceConfig` could use some quick/short doc strings for each parameter!